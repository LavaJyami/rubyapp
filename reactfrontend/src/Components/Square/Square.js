import React, { Component } from 'react';
import '../../App.css';

class Square extends Component {
  render(){
    var className = this.props.className;
    return(
      <div className={className} data-testid = "square">
        {this.props.value}
      </div>
    );
  }
}

export default Square;
