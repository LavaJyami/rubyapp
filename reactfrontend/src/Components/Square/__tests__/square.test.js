import React from 'react';
import ReactDOM from 'react-dom';
import Square from '../Square';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from 'react-test-renderer';


afterEach(cleanup);

it("Mainboard renders without crashing",() => {
  const div = document.createElement("div");
  ReactDOM.render(<Square value={"A"} />, div);
});

it("Letters in the square component are rendered correctly",() => {
  const {getByTestId} = render(<Square value = {"A"}/>);
  expect (getByTestId('square')).toHaveTextContent("A");
});

it("matches snapshot",  () => {
  const tree = renderer.create(<Square value = {"A"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
