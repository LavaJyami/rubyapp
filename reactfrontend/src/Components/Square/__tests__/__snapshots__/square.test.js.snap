// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="square"
  data-testid="square"
>
  A
</div>
`;
